### PR TITLE
Safer migration of gpxpod_tracks if a nightly > 5.0.0 was installed before

### PR DESCRIPTION
**Pull request by umgfoin**
There seems to be still an issue with latest schema-upgrade (5.0.3) which can break loading of new UI
The setter of entity "isEnabled" tries to bind to field "enabled" (although, I guess it should bind to new column "is_enabled").

{"reqId":"xxx","level":3,"time":"2022-11-15T16:51:02+00:00","remoteAddr":"::1","user":"asc","app":"index","method":"GET","url":"/apps/gpxpod/","message":"enabled is not a valid attribute","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:107.0) Gecko/20100101 Firefox/107.0","version":"25.0.1.1","exception":{"Exception":"BadFunctionCallException","Message":"enabled is not a valid attribute","Code":0,"Trace":[{"file":"/NEXTCLOUD_ROOT/lib/public/AppFramework/Db/Entity.php","line":165,"function":"setter","class":"OCP\\AppFramework\\Db\\Entity","type":"->"},{"file":"/NEXTCLOUD_ROOT/lib/public/AppFramework/Db/Entity.php","line":73,"function":"__call","class":"OCP\\AppFramework\\Db\\Entity","type":"->"},{"file":"/NEXTCLOUD_ROOT/lib/public/AppFramework/Db/QBMapper.php","line":322,"function":"fromRow","class":"OCP\\AppFramework\\Db\\Entity","type":"::"},{"file":"/NEXTCLOUD_ROOT/lib/public/AppFramework/Db/QBMapper.php","line":340,"function":"mapRowToEntity","class":"OCP\\AppFramework\\Db\\QBMapper","type":"->"},{"file":"/NEXTCLOUD_ROOT/apps/gpxpod/lib/Db/TrackMapper.php","line":97,"function":"findEntities","class":"OCP\\AppFramework\\Db\\QBMapper","type":"->"},{"file":"/NEXTCLOUD_ROOT/apps/gpxpod/lib/Controller/PageController.php","line":1084,"function":"getTracksOfUser","class":"OCA\\GpxPod\\Db\\TrackMapper","type":"->"},{"file":"/NEXTCLOUD_ROOT/apps/gpxpod/lib/Controller/PageController.php","line":197,"function":"cleanDbFromAbsentFiles","class":"OCA\\GpxPod\\Controller\\PageController","type":"->"},{"file":"/NEXTCLOUD_ROOT/lib/private/AppFramework/Http/Dispatcher.php","line":225,"function":"index","class":"OCA\\GpxPod\\Controller\\PageController","type":"->"},{"file":"/NEXTCLOUD_ROOT/lib/private/AppFramework/Http/Dispatcher.php","line":133,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/NEXTCLOUD_ROOT/lib/private/AppFramework/App.php","line":172,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/NEXTCLOUD_ROOT/lib/private/Route/Router.php","line":298,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/NEXTCLOUD_ROOT/lib/base.php","line":1047,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/NEXTCLOUD_ROOT/index.php","line":36,"function":"handleRequest","class":"OC","type":"::"}],"File":"/NEXTCLOUD_ROOT/lib/public/AppFramework/Db/Entity.php","Line":136,"CustomMessage":"--"}}

### CODE REVIEW

**Comment by julien-nc** [comment]
Thanks for the PR!
As this migration step will not run again on the next app upgrade (because it has run already :grin:) we can either:

* Just rename it and it will run again
* Create a new migration step removing the 'enabled' column

I would prefer the second option. What do you think?

**Comment by umgfoin** [comment]
> I would prefer the second option. What do you think?

Yes, I think that's preferable.

**Comment by umgfoin** [comment]
Should be OK now.

File: lib/Migration/Version050000Date20221116114851.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+                /** @var ISchemaWrapper $schema */
+                $schema = $schemaClosure();
+                $table = $schema->getTable('gpxpod_tracks');
+                if ($table->hasColumn('enabled') && $table->hasColumn('is_enabled') ) {
```
Comment (julien-nc) [inline lib/Migration/Version050000Date20221116114851.php]: Why checking if `is_enabled` is there as well? IMO we don't really care.
I would only check that `enabled` is there.

I'll merge and publish right after we agree on that :grin:.

Comment (umgfoin) [reply lib/Migration/Version050000Date20221116114851.php]: My approach: Change only, if schema meets your assumptions. But you're right:
- keeping the field name as is ('enabled') will have the same impact as ending up with neither of them
- after all this is to grease nightly-stuff..

**Review by julien-nc — APPROVED** [review]

**Comment by julien-nc** [comment]
@umgfoin As you're now a collaborator, you can push branches in this project instead of in your fork if you ever want to fix some more of my mistakes :grin:.

**Pull request merged**